### PR TITLE
Update DG for use cases

### PR DESCRIPTION
**Pull request by shaokiat**
- Refractor undo/redo to remove `c/`
- Update labelling for Appendix
- Added use cases for new features

[https://shaokiat.github.io/tp/DeveloperGuide.html](https://shaokiat.github.io/tp/DeveloperGuide.html)

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1402,7 +1469,7 @@ Test Cases:
    1. Open the jar file `gradpad.jar`<br>
    Expected: Saved data will reset to sample data and will be shown in the Completed Modules list.
 
-## **Appendix: Effort**
+## **Appendix D: Effort**
```
Comment (yan-soon) [inline docs/DeveloperGuide.md]: think this should be C instead hehe

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -765,37 +765,50 @@ When editing a module's tags, the old tags are simply "removed" as when
 
 #### Proposed Implementation
 
-The proposed undo/redo mechanism is facilitated by `VersionedGradPad`. It extends `GradPad` with an undo/redo history, stored internally as an `gradPadStateList` and `currentStatePointer`. Additionally, it implements the following operations:
+The proposed undo/redo mechanism is facilitated by `VersionedGradPad`. It extends `GradPad` with an undo/redo history,
+stored internally as an `gradPadStateList` and `currentStatePointer`. Additionally, it implements the
```
Comment (Silvernitro) [inline docs/DeveloperGuide.md]: NIt:
```suggestion
stored internally as a `gradPadStateList` and `currentStatePointer`. Additionally, it implements the 
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -946,66 +967,113 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2. GradPad displays the required modules in CS curriculum onto the `Result Display`
 
     Use case ends.
+    
+**Use case : UC08 - View all available General Education Modules offered in NUS**
+
+**MSS**
+
+1. User requests to view all available General Education Modules offered in NUS.
+2. GradPad displays the all available General Education modules onto the `Result Display`
```
Comment (yan-soon) [inline docs/DeveloperGuide.md]: extra "the" at the start

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -946,66 +967,113 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+    Use case ends.    
+
+**Use case : UC09 - View all available Science modules in the CS curriculum**
+
+**MSS**
+
+1. User requests to view all available Science modules in the CS curriculum.
+2. GradPad displays the all available Science modules in the CS curriculum onto the `Result Display`
```
Comment (yan-soon) [inline docs/DeveloperGuide.md]: same comment ^

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -882,41 +903,42 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**Use case : UC02 - Add a module into `Completed Modules`**
 
 **MSS**
 
 1. User requests to add a module into the `Completed Modules`
 2. GradPad adds the module into `Completed Modules`
-3. GradPad displays the module added onto the `Command Line Display`
+3. GradPad displays the module added onto the `Result Display`
 
     Use case ends.
```
Comment (Silvernitro) [inline docs/DeveloperGuide.md:1003]: I think we can add an extension when users try to add a module that doesn't exist in NUS

**Review by yan-soon — APPROVED** [review]
LGTM!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!1

**Review by Silvernitro — APPROVED** [review]

**Pull request merged**